### PR TITLE
Return Button and Page Title refactoring

### PR DESCRIPTION
**Pull request by ZachOBrien**
Making the return button turned into refactoring all the page titles as well. Before, they were a mix of different sized headers and all just baked into the template. Now to include a page title on a page you can just use the page title component so we don't have to worry about changing every single page if the design for the title changes. And this helps with the positioning of return button. Without this component you have to add divs and css to every single page. Again, not ideal. Here's what it looks like:

pagetitle-comp

your-teams

### CODE REVIEW

File: src/components/PageTitle.vue
Comment (liammoyn) [inline src/components/PageTitle.vue:20]: You can specify these are String props here too

File: src/components/ReturnButton.vue
Comment (liammoyn) [inline src/components/ReturnButton.vue:12]: Is this what we want? I think with this if you create a team (and are redirected to the single team view) if you press the back arrow you'd be redirected to the create form instead of the available teams as one might expect. That's just one example but I think we need the ability to pass in an optional parameter for a specifc view to go back to.

File: src/views/AvailableTeams.vue
Comment (liammoyn) [inline src/views/AvailableTeams.vue:13]: ? There's an h1 below for available teams

File: src/views/Settings.vue
Comment (liammoyn) [inline src/views/Settings.vue:56]: This should be the full name right?

File: src/views/TeamCreationView.vue
Comment (liammoyn) [inline src/views/TeamCreationView.vue:3]: Are the double quotes ("') here intentional?

File: src/components/ReturnButton.vue
Comment (ZachOBrien) [reply src/components/ReturnButton.vue:12]: Sure, we could do that. It just seemed natural to me from a usability standpoint that a back button takes you to the last page you were on. But I see that use case and that should be easy to add

File: src/views/AvailableTeams.vue
Comment (ZachOBrien) [reply src/views/AvailableTeams.vue:13]: Where do you see that? I don't.

File: src/views/Settings.vue
Comment (ZachOBrien) [reply src/views/Settings.vue:56]: fixed

File: src/views/TeamCreationView.vue
Comment (ZachOBrien) [reply src/views/TeamCreationView.vue:3]: What do you mean? The double quotes on :title=""? I don't know of any other way to pass data with props in this situation, and if you're passing a string directly you just wrap the inner ones in single quotes. Trying something like :title='Create a Team' will throw an error

File: src/components/PageTitle.vue
Comment (ZachOBrien) [reply src/components/PageTitle.vue:20]: updated

File: src/views/TeamCreationView.vue
Comment (liammoyn) [reply src/views/TeamCreationView.vue:3]: You have a `"` then a `'` right now. Should it just be a `"` or did you mean to also include the `'`?

File: src/components/ReturnButton.vue
Comment (liammoyn) [reply src/components/ReturnButton.vue:12]: Browsers already have a back button for this, the point of adding an additional one in our UI is so that we can go back to the last logical page rather than the exact last page they were on

File: src/views/AvailableTeams.vue
Comment (liammoyn) [reply src/views/AvailableTeams.vue:13]: Line 31 / 27

File: src/views/TeamCreationView.vue
Comment (ZachOBrien) [reply src/views/TeamCreationView.vue:3]: Yeah you need the '' to pass a String unless I missed some shortcut. Just doing :title="Create a Team" was giving me an error

**Comment by jackblanc** [comment]
What's the status of this one?

**Comment by jackblanc** [comment]
Screen Shot 2020-06-25 at 4 18 56 PM

**Comment by ZachOBrien** [comment]
> ICYMI, this is the build error. Can you take a look? Should be a quick fix.
> 
> Screen Shot 2020-06-25 at 4 18 56 PM

yep checking it out

File: src/views/AvailableTeams.vue
Comment (ZachOBrien) [reply src/views/AvailableTeams.vue:13]: thanks. Took it out

File: src/views/IndividualTeamView.vue
Comment (jackblanc) [inline src/views/IndividualTeamView.vue:7]: What should this actually say? Not "Test" I imagine.

Comment (ZachOBrien) [reply src/views/IndividualTeamView.vue:7]: Good catch, thanks. That's supposed to be the team name. Fixing now

**Comment by ZachOBrien** [comment]
> I think some changes in here are throwing mobile styling off. I don't know much about styling, but take a look at your changes on mobile view and see if you're also experiencing it.

Sharp eye. The issue was I was giving it a fixed width (makes it consistent across pages independent of how wide the parent div is) but that width was too wide for mobile. I added a media query to by default use a mobile-friendly width and then if on a screen where the wider width makes sense use the wider width

**Comment by jackblanc** [comment]
This looks much better. I'm still seeing issues on the /reserve/edit route, but I think it's probably unrelated. If you could investigate that'd be great. 

Also, I don't know if this is expected or not, but it seems like the individual team view always goes back to the current reservations page, and the current reservations page always goes back to the team view. I'm able to get stuck in a loop going back and forth between the two. Is that something we want?

**Comment by ZachOBrien** [comment]
> This looks much better. I'm still seeing issues on the /reserve/edit route, but I think it's probably unrelated. If you could investigate that'd be great.
> 
> Also, I don't know if this is expected or not, but it seems like the individual team view always goes back to the current reservations page, and the current reservations page always goes back to the team view. I'm able to get stuck in a loop going back and forth between the two. Is that something we want?
> 
> FYI, there's some merge conflicts now if you can please resolve those.
> 
> Thank you!

I fixed the styling on the reserve/edit route. Here are screenshots. I can change the button color back to green if you want, I don't remember changing that but might've accidentally taken a class off somewhere:

Old desktop view:
old-reserve-block-desktop

New desktop view:
new-reserve-block-desktop

Old mobile view:
old-reserve-block-mobile

New mobile view:
new-reserve-block-mobile

I also took out the route that was causing a loop from currentReservations. I'm not quite sure which pages we want to route back to specific pages and not simply the last page you were on, so I'll leave that to a discussion rather than guessing.

And merge conflict fixes in progress.

**Comment by ZachOBrien** [comment]
I'm trying to work in the hide/show header functionality now. I didn't realize that was there until just now fixing merge conflicts.

**Comment by liammoyn** [comment]
> I'm trying to work in the hide/show header functionality now. I didn't realize that was there until just now fixing merge conflicts.

I think it's okay if the back button is hidden when hiding the header if that's what you're worried about

**Comment by jackblanc** [comment]
@florisdobber @ZachOBrien what do we want to do with this PR?

File: src/components/Header.vue
Comment (liammoyn) [inline src/components/Header.vue:12]: We still good with this?

File: src/views/AvailableTeams.vue
Comment (liammoyn) [inline src/views/AvailableTeams.vue:5]: subtitle?

File: src/views/Settings.vue
Comment (liammoyn) [inline src/views/Settings.vue:109]: Why add this?

File: src/views/Settings.vue
Comment (liammoyn) [inline src/views/Settings.vue:116]: Where is this used?

File: src/views/MapView.vue
Comment (liammoyn) [inline src/views/MapView.vue:41]: Let's not commit comments into the mainline

File: src/views/MapView.vue
Comment (liammoyn) [inline src/views/MapView.vue:234]: No comments, just delete

File: src/views/MapView.vue
Comment (liammoyn) [inline src/views/MapView.vue:3]: @florisdobber / @willmt80 Seems like there are a lot of changes in this file, are these all what we want? Or did something get messed up from this PR sitting around for so long?

**Review by liammoyn — CHANGES_REQUESTED** [review]

File: src/components/Header.vue
Comment (florisdobber) [reply src/components/Header.vue:12]: Yes, I want this

File: src/views/MapView.vue
Comment (willmt80) [inline src/views/MapView.vue:37]: This block label button is not hooked up to anything and is a duplicate of the button already on the map. Do we want the button here?

File: src/views/MapView.vue
Comment (willmt80) [reply src/views/MapView.vue:3]: Most of the changes are pretty self contained, there are some name changes on an open PR but merging should be pretty easy.

File: src/views/AvailableTeams.vue
Comment (ZachOBrien) [reply src/views/AvailableTeams.vue:5]: this renders a computed property from a method called subtitle()

File: src/views/Settings.vue
Comment (ZachOBrien) [reply src/views/Settings.vue:109]: deleted

File: src/views/Settings.vue
Comment (ZachOBrien) [reply src/views/Settings.vue:116]: not used anymore. Deleted.

File: src/views/MapView.vue
Comment (ZachOBrien) [reply src/views/MapView.vue:37]: no. removed.

File: src/views/MapView.vue
Comment (ZachOBrien) [reply src/views/MapView.vue:41]: ok. removed this div and the buttons inside, they weren't being used.

File: src/views/MapView.vue
Comment (ZachOBrien) [reply src/views/MapView.vue:234]: deleted

**Review by liammoyn — APPROVED** [review]

File: src/views/MapView.vue
Comment (willmt80) [inline src/views/MapView.vue:89]: labelsVisible isn't used anywhere

**Pull request merged**
